model/task/task.go

```diff
@@ -3869,18 +3902,13 @@ func HasMatchingTasks(versionID string, opts HasMatchingTasksOptions) (bool, err
 
 func getTasksByVersionPipeline(versionID string, opts GetTasksByVersionOptions) []bson.M {
 	var match bson.M = bson.M{}
-
-	// Allow searching by either variant name or variant display
-	if len(opts.Variants) > 0 {
-		variantsAsRegex := strings.Join(opts.Variants, "|")
-
-		match = bson.M{
-			"$or": []bson.M{
-				{BuildVariantDisplayNameKey: bson.M{"$regex": variantsAsRegex, "$options": "i"}},
-				{BuildVariantKey: bson.M{"$regex": variantsAsRegex, "$options": "i"}},
-			},
-		}
+	if !opts.IncludeBuildVariantDisplayName && opts.UseLegacyAddBuildVariantDisplayName {
+		grip.Alert(message.Fields{
+			"message": "Implementer error: should not use UseLegacyAddBuildVariantDisplayName with !IncludeBuildVariantDisplayName",
```

Comment:
we should probably call this programmer error to be consistent with similar errors throughout the codebase.

Reply:
Thanks i knew there was another term for it but my grep skills weren't up to the task.